### PR TITLE
Remove basic tier from Studio pricing

### PR DESCRIPTION
**Pull request by mattseddon**
> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

-----

Related to https://github.com/iterative/studio/issues/9272.

This PR removes any mention of the previously available "basic" Studio pricing tier.

### CODE REVIEW

File: content/docs/sidebar.json [hunk 1]
```diff
@@ -730,14 +730,8 @@
               "remove-a-model-or-its-details"
             ]
           },
-          {
-            "slug": "team-collaboration",
-            "children": ["change-team-plan-and-size"]
-          },
-          {
-            "slug": "account-and-billing",
```
Comment (mattseddon) [inline content/docs/sidebar.json:738]: [F] account-and-billing has become account-management again. `billing-and-payment` has been deleted.

File: content/docs/sidebar.json [hunk 2]
```diff
@@ -730,14 +730,8 @@
               "remove-a-model-or-its-details"
             ]
           },
-          {
-            "slug": "team-collaboration",
-            "children": ["change-team-plan-and-size"]
```
Comment (mattseddon) [inline content/docs/sidebar.json:735]: [F] `change-team-plan-and-size` has been deleted. There is now a `Get Enterprise` section at the bottom of `team-collaboration`. All existing links should point to that section.

File: content/docs/studio/user-guide/team-collaboration/index.md [hunk 3]
```diff
@@ -179,22 +177,10 @@ seats. This means that you can invite 2 more collaborators. At this point, if
 you remove any one team member or pending invite, that seat becomes available
 and so you will have 3 remaining seats.
 
-### Change your team plan and team size
+## Get Enterprise
```
Comment (mattseddon) [inline content/docs/studio/user-guide/team-collaboration/index.md:180]: There were a few sections that mostly repeated this. We should be able to cut down to just this.

**Review by shcheklein — APPROVED** [review]

**Comment by mattseddon** [comment]
@dberenbaum PTAL and LMK/raise a PR if you want to change anything 🙏🏻.

**Pull request merged**